### PR TITLE
Ogs belowground 3d

### PR DESCRIPTION
**Pull request by jbathmann**
This PR adds the possibility to calculate belowground resource (re-)distribution by means of ogs.

### CODE REVIEW

File: Documentation/ProjectFile/MangaProject/tree_dynamics/belowground_competition/OGSLargeScale3D/abiotic_drivers/s_tide_daily_period.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+
+
+Optional: Period of daily tide. Default value is 12*60*60, i.e. the monthly tide has a cyclus of half a day.
```
Comment (zaphod-bx) [inline Documentation/ProjectFile/MangaProject/tree_dynamics/belowground_competition/OGSLargeScale3D/abiotic_drivers/s_tide_daily_period.md]: monthly raus

File: Documentation/ProjectFile/MangaProject/tree_dynamics/belowground_competition/OGSLargeScale3D/abiotic_drivers/s_tide_monthy_amplitude.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+
+
+Optional: Amplitude of monthly tide. Default value is .5, i.e. the monthly fluctuation of means varies between -.5 and .5 m around mean sea level (0m).
```
Comment (zaphod-bx) [inline Documentation/ProjectFile/MangaProject/tree_dynamics/belowground_competition/OGSLargeScale3D/abiotic_drivers/s_tide_monthy_amplitude.md]: dito

File: TreeModelLib/BelowgroundCompetition/OGSLargeScale3D/python_source.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        cell_id = self.cell_finder.FindCell([x, y, z])
+        return cell_id
+
+
+def constantContribution(cell_id):
+    return constant_contributions[cell_id]
+
+
+def salinityContribution(cell_id, salinity):
+    return salinity_prefactors[cell_id]
```
Comment (zaphod-bx) [inline TreeModelLib/BelowgroundCompetition/OGSLargeScale3D/python_source.py]: salinity_prefactor * salinity

File: TreeModelLib/BelowgroundCompetition/OGSLargeScale3D/OGSLargeScale3D.py [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+                                                geometry["h_stem"],
+                                                geometry["r_root"],
+                                                parameter["kf_sap"],
+                                                geometry["r_stem"])
+        R = root_surface_resistance + xylem_resistance
+        constant_contribution = -(
+            (parameter["leaf_water_potential"] +
+             (2 * geometry["r_crown"] + geometry["h_stem"]) * 9810) / R * 1000)
+        self._tree_constant_contribution.append(constant_contribution)
+        salinity_prefactor = -85000 * 1000 / R * 1000
```
Comment (zaphod-bx) [inline TreeModelLib/BelowgroundCompetition/OGSLargeScale3D/OGSLargeScale3D.py]: salinity_prefactor = -85000 * 1000 / R * 1000 * **density**

Comment (zaphod-bx) [inline TreeModelLib/BelowgroundCompetition/OGSLargeScale3D/OGSLargeScale3D.py]: something is very strange here

File: Documentation/ProjectFile/MangaProject/tree_dynamics/belowground_competition/OGSLargeScale3D/abiotic_drivers/s_tide_daily_period.md (hunk 1 above)
Comment (jbathmann) [reply Documentation/ProjectFile/MangaProject/tree_dynamics/belowground_competition/OGSLargeScale3D/abiotic_drivers/s_tide_daily_period.md]: fixed

File: TreeModelLib/BelowgroundCompetition/OGSLargeScale3D/python_source.py (hunk 3 above)
Comment (jbathmann) [reply TreeModelLib/BelowgroundCompetition/OGSLargeScale3D/python_source.py]: fixed

File: TreeModelLib/BelowgroundCompetition/OGSLargeScale3D/OGSLargeScale3D.py (hunk 4 above)
Comment (jbathmann) [reply TreeModelLib/BelowgroundCompetition/OGSLargeScale3D/OGSLargeScale3D.py]: revised

Comment (jbathmann) [reply TreeModelLib/BelowgroundCompetition/OGSLargeScale3D/OGSLargeScale3D.py]: revised

File: Documentation/ProjectFile/MangaProject/tree_dynamics/belowground_competition/OGSLargeScale3D/abiotic_drivers/s_tide_daily_period.md (hunk 1 above)
Comment (zaphod-bx) [reply Documentation/ProjectFile/MangaProject/tree_dynamics/belowground_competition/OGSLargeScale3D/abiotic_drivers/s_tide_daily_period.md]: okay

**Pull request merged**